src/ScalarMessaging.js

```diff
@@ -286,6 +286,30 @@ function inviteUser(event, roomId, userId) {
     });
 }
 
+/**
+ * Returns a promise that resolves when a widget with the given
+ * ID has been added as a user widget (ie. the accountData event
+ * arrives) or rejects after a timeout
+ */
+function waitForUserWidget(widgetId) {
+    return new Promise((resolve, reject) => {
+        let timerId;
+        function onAccountData(ev) {
+            if (ev.getContent()[widgetId] !== undefined) {
+                MatrixClientPeg.get().removeListener('accountData', onAccountData);
+                clearTimeout(timerId);
+                resolve();
+            }
+        }
+        timerId = setTimeout(() => {
+            console.log("Timed out waiting for widget ID " + widgetId + " to appear");
+            MatrixClientPeg.get().removeListener('accountData', onAccountData);
+            reject();
```

Comment:
If we reject with nothing, then we indicate that there's a failure to handle without indicating to the caller what the failure was. What about rejecting with an error and handling by logging from the caller?

Reply:
seems reasonable